### PR TITLE
[#40196] check and update the notifications api documentation

### PR DESCRIPTION
**Pull request by Kharonus**
- https://community.openproject.org/work_packages/40196
- renamed collection model
- reworked examples and object definitions
- small refactorings to remove linter warnings (#paths/projects_work_packages)

The file `docs/api/apiv3/paths/project_work_packages.yml` was only reworked to harmonize style and remove linter warnings. No functional changes were made.

### CODE REVIEW

**Review by cbliard — CHANGES_REQUESTED** [review]
There are things missing:
- api endpoints documentation to update read/unread status of notifications:
  - `post /api/v3/notifications/{id}/readIAN`
  - `post /api/v3/notifications/{id}/unreadIAN`
  - `post /api/v3/notifications/readIAN` (bulk update)
  - `post /api/v3/notifications/unreadIAN` (bulk update)
- links in `NotificationModel`
  - `unreadIAN` is missing, and is present when notification is read
  - `readIAN` is in required category, but in fact it will be absent if notification is read. Not sure how it should translate in openapi.

Apart from that, it looks good. I like that you ordered the keys in the yml to make the name of parameters first, or the description, summary, operationId and tags first. It makes the yml easier to read :+1:

**Comment by cbliard** [comment]
and I wonder if `docs/api/apiv3/tags/notifications.yml` should be updated too to add the `read` and `unread` links in the "Actions" section.

File: docs/api/apiv3/paths/notifications.yml [hunk 1]
```diff
@@ -1,23 +1,33 @@
 # /api/v3/notifications
 ---
 get:
+  summary: Get notification collection
+  operationId: Get_Notification_Collection
```
Comment (cbliard) [inline docs/api/apiv3/paths/notifications.yml]: For consistency with other operations on other resources, it should be `List_Notifications`

File: docs/api/apiv3/paths/notification.yml [hunk 2]
```diff
@@ -1,40 +1,36 @@
 # /api/v3/notifications/{id}
 ---
 get:
+  summary: Get the notification
+  operationId: Get_Notification
```
Comment (cbliard) [inline docs/api/apiv3/paths/notification.yml]: For consistency with other operation names, `operationId` should be `View_notification`

File: lib/api/v3/notifications/notification_representer.rb [hunk 3]
```diff
@@ -34,10 +34,7 @@ class NotificationRepresenter < ::API::Decorators::Single
         include API::Decorators::LinkedResource
         extend API::Decorators::PolymorphicResource
 
-        self_link title_getter: ->(*) { represented.subject }
-
         property :id
-        property :subject
```
Comment (cbliard) [inline lib/api/v3/notifications/notification_representer.rb:40]: This broke some unit tests testing for `subject` presence.

**Review by cbliard — CHANGES_REQUESTED** [review]

File: lib/api/v3/notifications/notification_representer.rb (hunk 3 above)
Comment (Kharonus) [reply lib/api/v3/notifications/notification_representer.rb:40]: fixed, and readded self link, but without title

File: docs/api/apiv3/paths/notification.yml (hunk 2 above)
Comment (Kharonus) [reply docs/api/apiv3/paths/notification.yml]: fixed

File: docs/api/apiv3/paths/notifications.yml (hunk 1 above)
Comment (Kharonus) [reply docs/api/apiv3/paths/notifications.yml]: fixed

**Comment by Kharonus** [comment]
@cbliard so, lets see, if I finally can satisfy you ;)

**Review by cbliard — APPROVED** [review]
Perfect 👌

**Pull request merged**